wgpu-core/src/track/texture.rs

```diff
@@ -218,16 +287,7 @@ impl Default for TextureUsageScope {
 impl TextureUsageScope {
     fn tracker_assert_in_bounds(&self, index: usize) {
         self.metadata.tracker_assert_in_bounds(index);
-
-        strict_assert!(index < self.set.simple.len());
-
-        strict_assert!(if self.metadata.contains(index)
```

Comment:
It does but it's already checked above in `self.metadata.tracker_assert_in_bounds(index);`.

Reply:
`ResourceMetadata::tracker_assert_in_bounds` does not actually assert that `index` is a member of the `ResourceMetadata, only that the entries in `owned` and `resources` are consistent.

Reply:
> `ResourceMetadata::tracker_assert_in_bounds` does not actually assert that `index` is a member of the `ResourceMetadata`

Doesn't the `strict_assert!(index < self.owned.len());` do that? `ResourceMetadata::contains` tries to lookup in `self.owned`.

Reply:
Teo and I talked about this in person, and we think this aspect of the PR is fine as is. The story is interesting, though.

1) The PR as it stands does indeed remove a check of the bit in `ResourceMetadata::owned` at `index`, so the condition being asserted is indeed affected, as Jim was concerned.

2) However, that condition had been redundant all along: When the bit in `owned` is clear, `self.set.simple[index]`, checked on the next line, will never be `TextureUses::COMPLEX`.

Reply:
> checked on the next line, will never be TextureUses::COMPLEX.

isn't this only the case if we actively clear the uses after deletion?

So this case may still be hit, though is a good argument for clearing actively.

Reply:
Given that we don't clear right now; is it an issue that with this PR we check `complex` contains the index regardless if `index` is owned in `metadata`?

I didn't think so because it would be inconsistent with the normal behavior of `TextureStateSet`. From what I found if `simple[index] == TextureUses::COMPLEX` then it should/will always be the case that `complex.contains_key(&index) == true`. I don't see why the check should be conditional on `self.metadata.contains(index)`.

Reply:
- We put texture of index A into the complex state.
- We delete the texture so `owned` is clear, but the usages is still COMPLEX
- We put a new texture into the tracker. This bounds check sees COMPLEX and thinks everything is groovy when its not?

Is that a problem?

Reply:
Discussed offline, this is fine.